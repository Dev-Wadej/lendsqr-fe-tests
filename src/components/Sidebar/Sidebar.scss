    .sidebar {
        font-family: 'Work Sans', sans-serif;
        overflow: scroll;
        width: 20vw;
        margin-top: 3rem;
        height: 90vh;
        box-shadow: 0px 5px 20px rgba(0, 0, 0, 0.04);
        padding-left: 2%;
        .sidebar-top {
            display: flex;
            gap: .7rem;
            span {
                color: rgba(33, 63, 125, 1);
                display: inline-block;
            }
        }
    }
    
    .sidebar-dashboard {
        margin-top: 4rem;
        p {
            color: #8d9cbd;
            font-size: .9rem;
            font-weight: 400;
        }
    }
    
    .sidebar-links {
        display: block;
        h4 {
            color: rgba(84, 95, 125, 1);
            font-size: .8rem;
            font-weight: 500;
            margin-top: 3rem;
            margin-bottom: 2rem;
        }
        a {
            span {
                display: inline-block;
            }
        }
    }
    
    .sidebar-links-each {
        display: flex;
        gap: .7rem;
        color: #717f9e;
        opacity: .7;
        font-weight: 400;
        font-size: 1rem;
        align-items: center;
        margin-bottom: 1.5rem;
    }
    
    .active-link {
        color: red;
    }
    
    #active {
        position: relative;
        .sidebar-links-eaches {
            color: #213f7d;
        }
    }
    
    #active::before {
        position: absolute;
        content: '';
        border-left: 2px solid #39CDCC;
        padding: 1rem;
        top: -.4rem;
        bottom: 0;
        left: -1.7rem;
        width: 15.8rem;
        background-color: #e5f6f6;
        z-index: -20;
    }