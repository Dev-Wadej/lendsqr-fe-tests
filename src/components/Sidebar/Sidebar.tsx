import React,{useEffect,useState} from 'react'
import {ReactComponent as ArrowDown} from '../../data/svgs/arrowdown2.svg'
import {ReactComponent as Briefcase} from '../../data/svgs/briefcase.svg'
import {ReactComponent as Home} from '../../data/svgs/home.svg'
import { customers,businesses,settings } from '../../data/dummydata'
import CloseIcon from '@mui/icons-material/Close';
import { useSidebar } from '../../context/ContextProvider' 
import './Sidebar.scss'
import { NavLink } from 'react-router-dom'

const Sidebar = () => {
    const {sidebar,setSidebar,setScreenSize,screenSize}=useSidebar()
    const [closeBtn,setCloseBtn]=useState(true)
    
    const handleClick=()=>{
        setSidebar?.(false)
    }
    useEffect(()=>{

            if(screenSize !==null){
                screenSize!<1000 ? setSidebar?.(false) :setSidebar?.(true)
                console.log('got here')
            }

            const handleResize=()=>setScreenSize?.(window.innerWidth);
            window.addEventListener('resize',handleResize)
            handleResize()
            return ()=> window.removeEventListener('resize',handleResize)
    
        },[screenSize, setScreenSize, setSidebar])

        useEffect(()=>{
            if(screenSize!==null){
                if(screenSize! > 1000  ){
                        setCloseBtn(false)
                }
                if(screenSize! <1000  ){
                    setCloseBtn(true)
            }
            }
        },[screenSize,sidebar])
  return(
      <>
      {sidebar && ( <section className='sidebar'>
    <div className='close' onClick={handleClick}>
        {
            closeBtn &&

<CloseIcon />
        }
</div>
<div className='sidebar-top'>
  <span><Briefcase/></span>
  <span>Switch Organization</span>
  <span><ArrowDown/></span>
</div>
<div className='sidebar-dashboard sidebar-top'>
  <span><Home/></span>
  <p>Dashboard</p>
</div>
<div className='sidebar-links'>
<h4 >{customers.title.toUpperCase()}</h4>
{customers.links.map((customerslink,idx)=>(
  <NavLink key={idx} className={'sidebar-links-each'} id={customerslink?.id} to={'#'}> 
       <span><customerslink.icons/></span>
       <span className='sidebar-links-eaches'>{customerslink.name}</span>
  </NavLink>
))}
</div>

<div className='sidebar-links'>
<h4 >{businesses.title.toUpperCase()}</h4>
{businesses.links.map((businesseslink,idx)=>(
  <NavLink key={idx} className={'sidebar-links-each'}  to={'#'}> 
       <span><businesseslink.icons/></span>
       <span className='sidebar-links-eaches'>{businesseslink.name}</span>
  </NavLink>
))}
</div>

<div className='sidebar-links'>
<h4 >{settings.title.toUpperCase()}</h4>
{settings.links.map((settingslink,idx)=>(
  <NavLink key={idx} className={'sidebar-links-each'}  to={'#'}> 
       <span><settingslink.icons/></span>
       <span className='sidebar-links-eaches'>{settingslink.name}</span>
  </NavLink>
))}
</div>
</section>)}
      </>
    
  )
    
    
  
}

export default Sidebar
