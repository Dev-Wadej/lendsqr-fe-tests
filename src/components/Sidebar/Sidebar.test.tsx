import Sidebar from './Sidebar'
import renderer from 'react-test-renderer';
import {cleanup, fireEvent, render} from '@testing-library/react';



afterEach(cleanup)
test('Sidebar component snapshot HTML', () => {
  const output = renderer.create(<Sidebar/>).toJSON();
  expect(output).toMatchSnapshot();
});