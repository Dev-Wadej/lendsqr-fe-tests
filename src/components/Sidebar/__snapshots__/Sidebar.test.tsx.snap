// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component snapshot HTML 1`] = `null`;
